Tests/Layout/RestrictedBrickSizeTest.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  RestrictedBrickSizeTest.swift
+//  BrickKit
+//
+//  Created by Peter Cheung on 7/28/17.
+//  Copyright © 2017 Wayfair. All rights reserved.
+//
+
+import XCTest
+@testable import BrickKit
+
+class RestrictedBrickSizeTest: XCTestCase {
+    var brickCollectionView: BrickCollectionView!
+    
+    override func setUp() {
+        super.setUp()
+        
+        brickCollectionView = BrickCollectionView(frame: CGRect(x: 0, y: 0, width: 320, height: 480))
+    }
+    
+    func testChangeBrickSizeGreaterThanMinimum() {
+        
+        let section = BrickSection("Test Section", bricks: [
+            DummyResizableBrick("Brick", height: .restricted(restriction: RestrictedBrickSize(size: .auto(estimate: .fixed(size: 100)), restrictedSize: .minimumSize(size: .fixed(size: 100)))), setHeight: 200)
+            ])
+        
+        brickCollectionView.setupSectionAndLayout(section)
+        
+        let cell = brickCollectionView.cellForItem(at: IndexPath(item: 0, section: 1)) as? DummyResizableBrickCell
+        cell?.layoutIfNeeded()
+        XCTAssertEqual(cell?.frame.size.height, 200)
+    }
+    
+    func testChangeBrickSizeLessThanMinimum() {
+        let section = BrickSection("Test Section", bricks: [
+            DummyResizableBrick("Brick", height: .restricted(restriction: RestrictedBrickSize(size: .auto(estimate: .fixed(size: 100)), restrictedSize: .minimumSize(size: .fixed(size: 100)))), setHeight: 50)])
+        
+        brickCollectionView.setupSectionAndLayout(section)
+        
+        let cell = brickCollectionView.cellForItem(at: IndexPath(item: 0, section: 1)) as? DummyResizableBrickCell
+        cell?.layoutIfNeeded()
+        XCTAssertEqual(cell?.frame.size.height, 100)
+    }
+    
+    func testChangeBrickSizeGreaterThanMaximum() {
+        let section = BrickSection("Test Section", bricks: [
+            DummyResizableBrick("Brick", height: .restricted(restriction: RestrictedBrickSize(size: .auto(estimate: .fixed(size: 100)), restrictedSize: .maximumSize(size: .fixed(size: 100)))), setHeight: 200)])
+        
+        brickCollectionView.setupSectionAndLayout(section)
+        
+        let cell = brickCollectionView.cellForItem(at: IndexPath(item: 0, section: 1)) as? DummyResizableBrickCell
+        cell?.layoutIfNeeded()
+        XCTAssertEqual(cell?.frame.size.height, 100)
+    }
+    
+    func testChangeBrickSizeLessThanMaximum() {
+        let section = BrickSection("Test Section", bricks: [
+            DummyResizableBrick("Brick", height: .restricted(restriction: RestrictedBrickSize(size: .auto(estimate: .fixed(size: 100)), restrictedSize: .maximumSize(size: .fixed(size: 100)))), setHeight: 50)])
+        
+        brickCollectionView.setupSectionAndLayout(section)
+        
+        let cell = brickCollectionView.cellForItem(at: IndexPath(item: 0, section: 1)) as? DummyResizableBrickCell
+        cell?.layoutIfNeeded()
+        XCTAssertEqual(cell?.frame.size.height, 50)
+    }
```

Comment:
It should still work with indirect dimensions. The error `fatalError("Only Ratio, Fixed and Fill are allowed")` should never happen